pymove/visualization/matplotlib.py

```diff
@@ -0,0 +1,199 @@
+import matplotlib.pyplot as plt
+
+from pymove.utils import constants
+from pymove.utils.constants import (
+    COUNT,
+    DATE,
+    DATETIME,
+    DAY,
+    HOUR,
+    LATITUDE,
+    LONGITUDE,
+    PERIOD,
+    SITUATION,
+    STOP,
+    TILES,
+    TRAJ_ID,
+)
+
+
+def show_object_id_by_date(
+    move_data,
+    create_features=True,
+    kind=None,
+    figsize=(21, 9),
+    return_fig=True,
+    save_fig=True,
+    name='shot_points_by_date.png',
+):
+    """Generates four visualizations based on datetime feature:
+
+        - Bar chart trajectories by day periods
+        - Bar chart trajectories day of the week
+        - Line chart trajectory by date
+        - Line chart of trajectory byhours of the day.
+
+    Parameters
+    ----------
+    move_data : pymove.core.MoveDataFrameAbstract subclass.
+        Input trajectory data.
+    create_features : bool, optional, default True.
+        Represents whether or not to delete features created for viewing.
+    kind: list or None
+        Determines the kinds of each plot
+    figsize : tuple, optional, default (21,9).
+        Represents dimensions of figure.
+    return_fig : bool, optional, default True.
+        Represents whether or not to save the generated picture.
+    save_fig : bool, optional, default True.
+        Represents whether or not to save the generated picture.
+    name : String, optional, default 'shot_points_by_date.png'.
+        Represents name of a file.
+
+    Returns
+    -------
+    fig : matplotlib.pyplot.figure or None
+        The generated picture.
+    References
+    ----------
+    https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.plot.html
+    """
+    if kind is None:
+        kind = ['bar', 'bar', 'line', 'line']
+
+    fig, ax = plt.subplots(2, 2, figsize=figsize)
+
+    move_data.generate_date_features()
+    move_data.generate_hour_features()
+    move_data.generate_time_of_day_features()
+    move_data.generate_day_of_the_week_features()
+
+    move_data.groupby([PERIOD])[TRAJ_ID].nunique().plot(
+        subplots=True, kind=kind[0], rot=0, ax=ax[0][0], fontsize=12
+    )
+    move_data.groupby([DAY])[TRAJ_ID].nunique().plot(
+        subplots=True, kind=kind[1], ax=ax[0][1], rot=0, fontsize=12
+    )
+    move_data.groupby([DATE])[TRAJ_ID].nunique().plot(
+        subplots=True,
+        kind=kind[2],
+        grid=True,
+        ax=ax[1][0],
+        rot=90,
+        fontsize=12,
+    )
+    move_data.groupby([HOUR])[TRAJ_ID].nunique().plot(
+        subplots=True, kind=kind[3], grid=True, ax=ax[1][1], fontsize=12
+    )
+
+    if not create_features:
+        move_data.drop(columns=[DATE, HOUR, PERIOD, DAY], inplace=True)
+
+    if save_fig:
+        plt.savefig(fname=name, fig=fig)
+
+    if return_fig:
+        return fig
+
+
+def show_lat_lon_gps(
+    move_data,
+    kind='scatter',
+    figsize=(21, 9),
+    plot_start_and_end=True,
+    return_fig=True,
+    save_fig=False,
+    name='show_gps_points.png',
+):
+    """
```

Comment:
Codacy found an issue: [Multi-line docstring summary should start at the first line (D212)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5714604)